storage/namespace/dynamic/registry.go

```diff
@@ -181,6 +182,12 @@ func (r *dynamicRegistry) run() {
 		}
 
 		val := r.kvWatch.Get()
+		if val == nil || reflect.ValueOf(val).IsNil() {
```

Comment:
Hm the way we've previously been protecting against an interface with nil ptr value is to make sure to return true `nil` instead of an interface with nil ptr for instance at the callsite.  e.g. see this type safe watch wrapping a `m3x/watch.Watch`:
https://github.com/m3db/m3db/blob/3d1c61958b68e0456c6d4936abb8176c9d710ba5/topology/map.go#L149-L155

Can't we do the same instead of using `reflect` here?

Reply:
Yep, so this arises because I've got a stub implementation of `kv.ValueWatch` (same file at the bottom) which doesn't do this. I'll see if I can use the standard implementation or make the fix in my stub impl. 

Either way, won't need the reflect check.

Reply:
Done. I replaced my stub implementation with `kv.ValueWatchable`, and removed the reflect call.